### PR TITLE
Rewrite container_top tests

### PR DESCRIPTION
**Pull request by apostasie**
On top of #3617

### CODE REVIEW

**Comment by apostasie** [comment]
Failure is connectivity issue with ECR

```failed to do request: Head \"[https://public.ecr.aws/v2/soci-workshop-examples/ffmpeg/manifests/latest\](https://public.ecr.aws/v2/soci-workshop-examples/ffmpeg/manifests/latest/)": net/http: TLS handshake timeout" host=public.ecr.aws
        time="2024-10-30T19:57:47Z" level=fatal msg="failed to resolve reference \"public.ecr.aws/soci-workshop-examples/ffmpeg:latest\": failed to do request: Head \"[https://public.ecr.aws/v2/soci-workshop-examples/ffmpeg/manifests/latest\](https://public.ecr.aws/v2/soci-workshop-examples/ffmpeg/manifests/latest/)": net/http: TLS handshake timeout"```

File: cmd/nerdctl/container/container_commit_test.go [hunk 1]
```diff
@@ -36,14 +36,7 @@ func TestCommit(t *testing.T) {
 				helpers.Anyhow("rmi", "-f", data.Identifier())
 			},
 			Setup: func(data test.Data, helpers test.Helpers) {
-				// FIXME: short of pulling first, docker will fail to start the container.
-				// Debugging shows container exited immediately. Nothing in the container logs. Not much in journalctl.
-				// It is not clear what is happening.
-				if nerdtest.IsDocker() {
-					helpers.Ensure("pull", testutil.CommonImage)
-				}
```
Comment (AkihiroSuda) [inline cmd/nerdctl/container/container_commit_test.go:44]: This change doesn't seem relevant to top?

Comment (apostasie) [reply cmd/nerdctl/container/container_commit_test.go:44]: Correct.
top_test was suffering from the same issue, which I fixed in a separate PR, and I ended-up removing both workarounds at the same time.
Lmk of you want another commit for this.

Comment (AkihiroSuda) [reply cmd/nerdctl/container/container_commit_test.go:44]: Yes, another commit is more preferable

Comment (apostasie) [reply cmd/nerdctl/container/container_commit_test.go:44]: Done.

**Review by AkihiroSuda — APPROVED** [review]
Thanks
